tools/run_tests/artifacts/distribtest_targets.py

```diff
@@ -476,35 +476,40 @@ def targets():
         PythonDistribTest("linux", "x64", "ubuntu2404", source=True),
         # Ruby
         RubyDistribTest(
-            "linux",
+            "linux-gnu",
```

Comment:
Hmm latest distribtests failed with a new error, which is that  the `linux` we're passing to `task_runner.py` in [here](https://github.com/grpc/grpc/blob/636edd060af1416dc9c57b9cb8d7768f10d0c44c/tools/internal_ci/linux/grpc_distribtests_ruby.sh#L53) did not pick up any tests (since we've changed the `platform` field of these distribtest targets.

Maybe augment the way we generate labels:

```
self.labels = ["distribtest", "ruby", platform, arch, docker_suffix]
```

to something like this:

```
platform_label = self.platform
if platform_label.startswith('linux')
  platform_label = 'linux'
self.labels = ["distribtest", "ruby", platform_label, arch, docker_suffix]
```

so that we pick up all linux sub-variants with the linux label